src/site/_includes/components/cookie/cookie.ts

```diff
@@ -0,0 +1,41 @@
+export function getCookie(name: string) {
+  const nameEQ = `${name}=`
+  const cookies = document.cookie.split('; ')
+  for (let i = 0, len = cookies.length; i < len; i += 1) {
+    const cookie = cookies[i]
+    if (cookie.indexOf(nameEQ) === 0) {
+      return decodeURIComponent(cookie.substring(nameEQ.length))
+    }
+  }
+  return null
+}
+
+/**
+ * Set the cookie
+ * @param name - The name of the cookie
+ * @param value - The value of the cookie
+ * @param ttl - The time to live in days
+ */
+export function setCookie(name: string, value: string, ttl: number) {
+  let cookieString = `${name}=${value}; path=/`
+  const date = new Date()
+  date.setTime(date.getTime() + (ttl * 24 * 60 * 60 * 1000))
+  cookieString = `${cookieString}; expires=${date.toUTCString()}`
+
+  if (document.location.protocol === 'https:') {
+    cookieString = `${cookieString}; Secure`
+  }
+  document.cookie = cookieString
+}
+
+export function deleteCookie(name: string) {
+  if (getCookie(name)) {
+    // Cookies need to be deleted in the same level of specificity in which they were set
+    // If a cookie was set with a specified domain, it needs to be specified when deleted
+    // If a cookie wasn't set with the domain attribute, it shouldn't be there when deleted
+    // You can't tell if a cookie was set with a domain attribute or not, so try both options
+    document.cookie = `${name}=;expires=Thu, 01 Jan 1970 00:00:00 GMT;path=/`
+    document.cookie = `${name}=;expires=Thu, 01 Jan 1970 00:00:00 GMT;domain=${window.location.hostname};path=/`
+    document.cookie = `${name}=;expires=Thu, 01 Jan 1970 00:00:00 GMT;domain=.${window.location.hostname};path=/`
+  }
+}
```

Comment:
This was taken from https://github.com/alphagov/govuk-design-system/blob/65d74bb7f463462f8b5773a21c412704578cd61e/src/javascripts/components/cookie-functions.js. and simplified